### PR TITLE
Add documentation popup for code completions

### PR DESCRIPTION
**Pull request by vlaaad**
User-facing changes:

The code editor will now show a documentation popup for code completions if you press <kbd>Ctrl Space</kbd> again when the completion popup is already shown. The documentation popup shows the same information in the [Defold reference](https://defold.com/ref/stable/go/): descriptions, parameters, returns, and examples. It renders markdown, which will help integrate LSP completions that usually present the documentation using markdown formatting.

https://github.com/defold/defold/assets/732554/21ed7fe4-3d04-4a64-a648-0f4f78260e1b

Technical notes:

I ported the completion popup to cljfx. I also tidied up the positioning and sizing of the completion and documentation popups. The viewer is implemented as a Defold-flavored HTML cljfx view that uses VBoxes and TextFlows, where Markdown is transformed into HTML. Took me a while to get right :D

Related to #7699

### CODE REVIEW

**Comment by AGulev** [comment]
>The code editor will now show a documentation popup for code completions if you press Ctrl Space again when the completion popup is already shown. 

Could you pls add it into the editor manual in our `doc` repo? Also, it would be nice to make an announcement on the forum.

File: editor/src/clj/editor/markdown.clj [hunk 1]
```diff
@@ -13,12 +13,398 @@
...
+(def ^:private ext-with-pref-height-defining-max-width
+  (fx/make-ext-with-props
+    {:max-width (fx.prop/make
+                  (fx.mutator/setter
+                    (fn [^ScrollPane pane [_key max-width]]
+                      (.setMaxWidth pane max-width)
+                      (.applyCss pane)
+                      (.setPrefViewportHeight pane (.prefHeight (.getContent pane) max-width))
+                        ;; this is needed in some rare cases T_T
+                        ;; e.g. type "vmath", then type "."
```
Comment (matgis) [inline editor/src/clj/editor/markdown.clj:357]: Style: The indentation of this comment looks a bit off?

**Review by matgis — CHANGES_REQUESTED** [review]
Works great! The markdown mangling looks like it could have been painful. 😅

I found some reflection warnings and one place where I'm concerned about a `g/with-auto-evaluation-context` block lifetime.

File: editor/src/clj/editor/code/view.clj [hunk 2]
```diff
@@ -1403,6 +1430,217 @@
...
+                                       10.0                    ;; scroll bar
+                                       0.0)
+                                     (->> completions
+                                          (eduction
+                                            (take 512)
+                                            (map #(-> text
+                                                      (doto (.setText (:display-string %)))
+                                                      .getLayoutBounds
+                                                      .getWidth)))
+                                          ^double (reduce max 0.0)
```
[the remark this answers is not in the document]
Comment (vlaaad) [reply editor/src/clj/editor/code/view.clj:1506]: Yes!

File: editor/src/clj/editor/code/view.clj [hunk 3]
```diff
@@ -60,16 +68,16 @@
...
            [javafx.scene.canvas Canvas GraphicsContext]
-           [javafx.scene.control Button CheckBox PopupControl Tab TextField]
+           [javafx.scene.control Button CheckBox Tab TextField]
            [javafx.scene.input Clipboard DataFormat InputMethodEvent InputMethodRequests KeyCode KeyEvent MouseButton MouseDragEvent MouseEvent ScrollEvent]
            [javafx.scene.layout ColumnConstraints GridPane Pane Priority]
            [javafx.scene.paint Color LinearGradient Paint]
            [javafx.scene.shape Rectangle]
            [javafx.scene.text Font FontSmoothingType Text TextAlignment]
-           [javafx.stage Stage]))
+           [javafx.stage PopupWindow Screen Stage]))
```
[the remark this answers is not in the document]
Comment (vlaaad) [reply editor/src/clj/editor/code/view.clj:80]: yes!

**Pull request merged**